src/modules/75-cf/pages/feature-flags-detail/flag-pipeline-tab/components/execution-list/execution-card/targeting/PercentageRolloutItem.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { Color, FontVariation } from '@harness/design-system'
+import { Container, Text, Icon } from '@harness/uicore'
+import React from 'react'
+import useFlagVariation from '@cf/pages/feature-flags-detail/flag-pipeline-tab/hooks/useFlagVariation'
+import type { Distribution, ServingRule, Variation } from 'services/cf'
+import { useStrings } from 'framework/strings'
+import css from '../../ExecutionList.module.scss'
+
+interface PercentageRolloutItemProps {
+  percentageRollout: ServingRule
+  flagVariations: Variation[]
+}
+
+const PercentageRolloutItem: React.FC<PercentageRolloutItemProps> = ({ percentageRollout, flagVariations }) => {
+  const { getString } = useStrings()
+
+  const { getVariationNameById, getVariationColorById } = useFlagVariation({
+    flagVariations
+  })
+
+  return (
+    <>
+      <span className={css.targetingItem}>
+        <Text font={{ variation: FontVariation.SMALL }} color={Color.GREY_900} inline>
+          {getString('cf.pipeline.flagConfiguration.serve')}
+        </Text>
+        <Text font={{ variation: FontVariation.SMALL_BOLD }} color={Color.GREY_900} inline>
+          {getString('cf.featureFlags.percentageRollout')}
+        </Text>
+        <Text font={{ variation: FontVariation.SMALL }} color={Color.GREY_900} inline>
+          {getString('cf.featureFlags.flagPipeline.to')}
+        </Text>
+        <Text
+          font={{ variation: FontVariation.SMALL_BOLD }}
+          color={Color.GREY_900}
+          inline
+          data-testid={`percentage-rollout-group`}
+        >
+          {percentageRollout.clauses[0].values[0]}
+        </Text>
+      </span>
+
+      <Container className={css.percentageRolloutVariations}>
+        <>
+          {(percentageRollout.serve.distribution as Distribution).variations.map(percentageRolloutVariation => (
+            <Container flex={{ alignItems: 'center' }} key={percentageRolloutVariation.variation}>
+              <Icon
+                name="full-circle"
+                size={12}
+                style={{
+                  color: getVariationColorById(percentageRolloutVariation.variation)
+                }}
+                margin={{ right: 'xsmall' }}
+              />
+              <Text
+                font={{ variation: FontVariation.SMALL }}
+                color={Color.GREY_500}
+                inline
+                data-testid="percentage-rollout-variation-weight"
+              >
+                {getVariationNameById(percentageRolloutVariation.variation)} ({percentageRolloutVariation.weight}%)
+              </Text>
+            </Container>
```

Comment:
I think `Text` supports an `icon` prop. 
```suggestion
            <Text
              key={percentageRolloutVariation.variation}>
              icon="full-circle"
              iconProps={{
                size: 10,
                color: getVariationColorById(percentageRolloutVariation.variation)
              }}
              font={{ variation: FontVariation.SMALL }}
              color={Color.GREY_500}
              data-testid="percentage-rollout-variation-weight"
            >
              {getVariationNameById(percentageRolloutVariation.variation)} ({percentageRolloutVariation.weight}%)
            </Text>
```

Reply:
Usually yes - except it doesn't work - the icons appear gray 😢

I _think_ its because our variation colors are hex strings as opposed to `Color` type? Not sure, if you've any guidance on this I'm all ears 👍  

<img width="324" alt="Screenshot 2022-08-04 at 13 45 56" src="https://user-images.githubusercontent.com/10871707/182850388-9f5ae823-7854-4aa9-8440-b67665bff603.png">

Reply:
But it's a lovely shade of grey! You can use the `style` prop in `iconProps`
```suggestion
            <Text
              key={percentageRolloutVariation.variation}>
              icon="full-circle"
              iconProps={{
                size: 10,
                style: {
                  color: getVariationColorById(percentageRolloutVariation.variation)
                }
              }}
              font={{ variation: FontVariation.SMALL }}
              color={Color.GREY_500}
              data-testid="percentage-rollout-variation-weight"
            >
              {getVariationNameById(percentageRolloutVariation.variation)} ({percentageRolloutVariation.weight}%)
            </Text>
```

Reply:
ah success! thanks!